electrum/lnsweep.py

```diff
@@ -328,35 +337,107 @@ def analyze_ctx(chan: 'Channel', ctx: Transaction):
     return ctn, their_pcp, is_revocation, per_commitment_secret
 
 
+def extract_funding_pubkeys_from_ctx(txin: TxInput) -> Tuple[bytes, bytes]:
+    """Extract the two funding pubkeys from the published commitment transaction.
+
+    We expect to see a witness script of: OP_2 pk1 pk2 OP_2 OP_CHECKMULTISIG"""
+    elements = txin.witness_elements()
+    witness_script = elements[-1]
+    assert match_script_against_template(witness_script, SCRIPTPUBKEY_TEMPLATE_FUNDING)
```

Comment:
nit. `SCRIPTPUBKEY_TEMPLATE_FUNDING` as a name is a bit confusing:
"scriptPubKey" is conventionally the name for the output script directly visible in the chain (i.e. the script in the output, so what a bitcoin address directly maps to).

So e.g. these are indeed templates for scriptPubKeys:
https://github.com/spesmilo/electrum/blob/437a9e435847431e8654a6e02fc304a714826533/electrum/transaction.py#L436-L442
The templates here are also misnamed IMHO, as they are matching witnessScripts (note difference between witness and witnessScript):
https://github.com/spesmilo/electrum/blob/437a9e435847431e8654a6e02fc304a714826533/electrum/submarine_swaps.py#L37

Maybe just call it `SCRIPT_TEMPLATE_FUNDING`?

Reply:
Makes sense, done.